legend-sdlc-server-fs/src/main/java/org/finos/legend/sdlc/server/api/workspace/FileSystemWorkspaceApi.java

```diff
@@ -0,0 +1,339 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.sdlc.server.api.workspace;
+
+import org.eclipse.collections.api.factory.Lists;
+import org.eclipse.collections.api.list.MutableList;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.ListBranchCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.Constants;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.finos.legend.sdlc.domain.model.project.workspace.Workspace;
+import org.finos.legend.sdlc.domain.model.project.workspace.WorkspaceType;
+import org.finos.legend.sdlc.server.api.BaseFSApi;
+import org.finos.legend.sdlc.server.api.entity.FileSystemEntityApi;
+import org.finos.legend.sdlc.server.api.user.FileSystemUserApi;
+import org.finos.legend.sdlc.server.domain.api.workspace.*;
+import org.finos.legend.sdlc.server.error.LegendSDLCServerException;
+import org.finos.legend.sdlc.server.exception.UnavailableFeature;
+import org.finos.legend.sdlc.server.project.ProjectFileAccessProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.ws.rs.core.Response;
+
+import static org.finos.legend.sdlc.server.domain.api.project.source.SourceSpecification.workspaceSourceSpecification;
+
+public class FileSystemWorkspaceApi extends BaseFSApi implements WorkspaceApi
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileSystemWorkspaceApi.class);
+
+    private static final String WORKSPACE_BRANCH_PREFIX = "workspace";
+    private static final String CONFLICT_RESOLUTION_BRANCH_PREFIX = "resolution";
+    private static final String BACKUP_BRANCH_PREFIX = "backup";
+    private static final String GROUP_WORKSPACE_BRANCH_PREFIX = "group";
+    private static final String GROUP_CONFLICT_RESOLUTION_BRANCH_PREFIX = "group-resolution";
+    private static final String GROUP_BACKUP_BRANCH_PREFIX = "group-backup";
+    private static final String PATCH_WORKSPACE_BRANCH_PREFIX = "patch";
+
+    protected static final char BRANCH_DELIMITER = '/';
+
+    @Inject
+    public FileSystemWorkspaceApi()
+    {
+    }
+
+    @Override
+    public Workspace getWorkspace(String projectId, WorkspaceSpecification workspaceSpecification)
+    {
+        String refBranchName = Constants.R_HEADS + FileSystemEntityApi.getRefBranchName(workspaceSourceSpecification(workspaceSpecification));
```

Comment:
This class should not be accessing static method on FileSystemEntityApi. The method probably belongs in a central utilities class or a shared super-class.